library/src/main/java/com/otaliastudios/zoom/ZoomEngine.kt

```diff
@@ -223,44 +175,28 @@ internal constructor(context: Context) : ViewTreeObserver.OnGlobalLayoutListener
     override val panY: Float
         get() = scaledPanY / realZoom
 
-    /**
-     * The current pan as a [ScaledPoint]
-     * This field will be updated according to current scaled pan when accessed.
-     *
-     * TODO if necessary, expose scaledPan, scaledPanX and scaledPanY through ZoomApi.
-     * Right now I'm not sure it makes sense - all input APIs work with absolute values and the
-     * "scaled pan" concept is probably not even mentioned in docs.
-     */
-    private val scaledPan = ScaledPoint()
-        get() {
-            field.set(scaledPanX, scaledPanY)
-            return field
-        }
+    //region MatrixManager delegates
 
-    /**
-     * The current horizontal scaled pan, which is the pan position of the content
-     * according to the current zoom value (so it's scaled).
-     */
+    // TODO needed? access through MatrixManager?
     @ScaledPan
-    private val scaledPanX: Float
-        get() = mContentScaledRect.left
+    internal val contentScaledWidth get() = matrixManager.contentScaledWidth
```

Comment:
It's just a matter of breaking the ZoomEngine monolith into testable components. In the future will create base interfaces for this components, so we can write tests.

I also wanted to bring this in before working on a new feature (rotation). I will refine this further and add docs 👍 

I wonder if, instead of keeping a `zoom` float, we could compute the current zoom by comparing `contentRect` and `contentScaledRect` (which we already use to compute pan), it would be much cleaner